src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -200,6 +200,12 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
             bool hasErrors,
             BindingDiagnosticBag diagnostics)
         {
+            if (hasSuppression(expression))
+            {
+                diagnostics.Add(ErrorCode.ERR_IllegalSuppression, expression.Location);
```

Comment:
:memo: This used to be a parsing error. We could decide to allow this (although not useful since we never produce nullability warnings in such location that would need suppression) but thought it safer to continue disallowing.